app/scripts/lib/rpc-method-middleware/handlers/add-ethereum-chain.test.js

```diff
@@ -85,7 +86,7 @@ describe('addEthereumChainHandler', () => {
   });
 
   describe('with `endowment:permitted-chains` permissioning inactive', () => {
-    it('creates a new network configuration for the given chainid and switches to it if none exists', async () => {
+    it('creates a new network configuration for the given chainid and switches to it for the dapp and globally if none exists and there are no pending requests before switching', async () => {
```

Comment:
I understand I made a little bit of a mess here by overloading this scenario, but I do have this entire test (and the switchEthereumChain tests) refactored in [this PR ](https://github.com/MetaMask/metamask-extension/pull/27847) which mocks out the switchChain() helper call and tests it separately. It took me an entire day to unravel these tests and add tests to cover switchChain() separately. I'd rather not try to clean this up here given this tech debt *will* be short lived